internal/cmd/deploy/deploy.go

```diff
@@ -0,0 +1,187 @@
+package deploy
+
+import (
+	"context"
+	"fmt"
+	"os/exec"
+	"strings"
+
+	"github.com/briandowns/spinner"
+	"github.com/google/go-github/github"
+	"github.com/spf13/cobra"
+	"github.com/zeabur/cli/internal/cmdutil"
+	"github.com/zeabur/cli/internal/util"
+)
+
+type Options struct {
+	name string
+}
+
+func NewCmdDeploy(f *cmdutil.Factory) *cobra.Command {
+	opts := &Options{}
+
+	cmd := &cobra.Command{
+		Use:     "deploy",
+		Short:   "Deploy a service",
+		PreRunE: util.NeedProjectContextWhenNonInteractive(f),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return runDeploy(f, opts)
+		},
+	}
+
+	cmd.Flags().StringVar(&opts.name, "name", "", "Service name")
+
+	return cmd
+}
+
+func runDeploy(f *cmdutil.Factory, opts *Options) error {
+	if f.Interactive {
+		return runDeployInteractive(f, opts)
+	} else {
+		return runDeployNonInteractive(f, opts)
+	}
+}
+
+func runDeployNonInteractive(f *cmdutil.Factory, opts *Options) error {
+	repoOwner, repoName, err := getRepoInfo()
+	if err != nil {
+		return err
+	}
+
+	repoID, err := getRepoID(repoOwner, repoName)
+	if err != nil {
+		return err
+	}
+
+	fmt.Println(repoID)
+
+	return nil
+}
+
+func runDeployInteractive(f *cmdutil.Factory, opts *Options) error {
+	s := spinner.New(cmdutil.SpinnerCharSet, cmdutil.SpinnerInterval,
+		spinner.WithColor(cmdutil.SpinnerColor),
+		spinner.WithSuffix(" Fetching repository informations..."),
+	)
+	s.Start()
+	repoOwner, repoName, err := getRepoInfo()
+	if err != nil {
+		return err
+	}
+
+	// Use repo name as default service name
+	if opts.name == "" {
+		opts.name = repoName
+	}
+
+	// Get repo ID and branches concurrently
```

Comment:
You can try to rewrite this code using `errgroup`.

Reply:
OK I'll try it later